src/UI/Art/FillRectArtist/FillRectMenuArtist.java

```diff
@@ -43,51 +44,39 @@ public void drawMenu(Graphics g)
     g.setColor(Color.cyan); // inner fill
     g.fillRect(menuBorderLeft + 1, menuBorderTop + 1, menuWidth - 2, menuHeight - 2);
     String label;
-    switch (myView.currentMenu.menuType)
+
+    InGameMenu<? extends Object> currentMenu = myView.getCurrentGameMenu();
+    if( null != currentMenu )
     {
-      case PRODUCTION:
-        g.setColor(COLOR_CURSOR);
-        g.fillRect(menuBorderLeft + 1, (myView.currentMenu.getSelectionNumber() + 1) * tileSizePx / 2 + menuBorderTop + 4,
-            menuWidth - 2, tileSizePx / 2);
-        g.setColor(Color.MAGENTA);
+      int yOffset = 1; // Helps to guide where we draw the menu options. Will offset further if we draw money first.
+      if( currentMenu.getSelectedOption() instanceof Units.UnitModel )
```

Comment:
Using `instanceof` here is slightly undesireable; we could add the `MenuType` enum back in instead, but that limits the extensibility of `InGameMenu` somewhat. Thoughts?